### ISSUE TITLE
Bootstrapping SLES4SAP12 SP5 fails: One or more requisite failed: bootstrap.salt-minion-package

### ISSUE THREAD

**Issue by stdevel** [uyuni-project/uyuni#2842]
I'm trying to bootstrap a SLES4SAP12 SP5 client with the UI and Salt but it fails. I also have other SLES4SAP machines that were bootstrapped successfully.

This issue seems to be similiar to #2740 - but there are no missing packages. See details below.

## Version of Uyuni Server and Proxy (if used)
`zypper info Uyuni-Server-release` _and_ `zypper info Uyuni-Proxy-release` _(if used)_

```
zypper info Uyuni-Server-release
Loading repository data...
Reading installed packages...


Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : uyuni-server-stable
Name           : Uyuni-Server-release
Version        : 2020.09-98.2.uyuni
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Server-release-2020.09-98.2.uyuni.src
Summary        : Uyuni Server
Description    :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.
```

## Details about issue

SLES4SAP 12 content were synchronized within the UI after enter the Organization Mirror credentials.

Creating the bootstrap repo works fine:

```
#  mgr-create-bootstrap-repo --with-custom-channels
1. SLES4SAP-12-SP2-x86_64
2. SLES4SAP-12-SP3-x86_64
3. SLES4SAP-12-SP4-x86_64
4. SLES4SAP-12-SP5-x86_64
Enter a number of a product label: 3
 
Creating bootstrap repo for SLES4SAP-12-SP4-x86_64
copy 'dbus-1-python-1.2.0-4.194.x86_64'
copy 'hwdata-0.314-2.4.uyuni.noarch'
copy 'libcurl4-7.60.0-4.15.2.x86_64'
copy 'libgudev-1_0-0-230-1.5.x86_64'
copy 'libnewt0_52-0.52.16-2.8.uyuni.x86_64'
copy 'libnl1-1.1.4-4.21.x86_64'
copy 'libopenssl1_0_0-1.0.2p-3.27.1.x86_64'
copy 'libsqlite3-0-3.8.10.2-9.15.1.x86_64'
copy 'libudev1-228-150.86.3.x86_64'
copy 'udev-228-150.86.3.x86_64'
copy 'openssl-1.0.2p-1.13.noarch'
copy 'python-libxml2-2.9.4-46.23.3.x86_64'
copy 'libzypp-16.21.2-2.45.1.x86_64'
copy 'newt-0.52.16-2.8.uyuni.x86_64'
copy 'python-2.7.17-28.45.1.x86_64'
copy 'python-cffi-1.11.2-5.11.1.x86_64'
copy 'python-cryptography-2.1.4-7.28.2.x86_64'
copy 'python-dmidecode-3.12.1-14.2.8.uyuni.x86_64'
copy 'python-ethtool-0.11-5.2.x86_64'
copy 'python-gobject2-2.28.6-24.80.x86_64'
copy 'python-gudev-147.2-2.1.uyuni.x86_64'
copy 'python2-hwdata-2.3.5-2.6.uyuni.noarch'
copy 'python-newt-0.52.16-2.8.uyuni.x86_64'
copy 'python-pyasn1-0.1.9-4.6.1.noarch'
copy 'python-pycparser-2.10-5.6.1.noarch'
copy 'python-pyOpenSSL-17.1.0-4.23.1.noarch'
copy 'python-six-1.11.0-9.23.2.uyuni.noarch'
copy 'python-xml-2.7.17-28.45.1.x86_64'
copy 'python-pyudev-0.16.1-3.2.1.noarch'
copy 'python2-rhnlib-4.2.1-2.1.uyuni.noarch'
copy 'rpm-python-4.11.2-16.21.1.x86_64'
copy 'libsolv-tools-0.6.36-2.30.1.x86_64'
copy 'libslang2-2.2.4-9.7.x86_64'
copy 'spacewalk-check-4.2.2-2.1.uyuni.noarch'
copy 'python2-spacewalk-check-4.2.2-2.1.uyuni.noarch'
copy 'spacewalk-client-setup-4.2.2-2.1.uyuni.noarch'
copy 'python2-spacewalk-client-setup-4.2.2-2.1.uyuni.noarch'
copy 'spacewalk-client-tools-4.2.2-2.1.uyuni.noarch'
copy 'python2-spacewalk-client-tools-4.2.2-2.1.uyuni.noarch'
copy 'python2-uyuni-common-libs-4.2.1-2.1.uyuni.x86_64'
copy 'mgr-daemon-4.2.2-2.1.uyuni.noarch'
copy 'suseRegisterInfo-4.2.1-2.1.uyuni.noarch'
copy 'python2-suseRegisterInfo-4.2.1-2.1.uyuni.noarch'
copy 'zypp-plugin-python-0.6.3-3.6.1.noarch'
copy 'zypp-plugin-spacewalk-1.0.7-3.1.uyuni.noarch'
copy 'python2-zypp-plugin-spacewalk-1.0.7-3.1.uyuni.noarch'
copy 'zypper-1.13.55-21.29.1.x86_64'
copy 'yast2-packager-3.2.30-2.13.12.x86_64'
copy 'yast2-pkg-bindings-3.2.5-2.6.36.x86_64'
copy 'python-backports.ssl_match_hostname-3.4.0.2-2.2.uyuni.noarch'
copy 'python-futures-3.0.5-9.2.uyuni.noarch'
copy 'python-Jinja2-2.8-19.19.3.uyuni.noarch'
copy 'python-MarkupSafe-0.23-16.8.5.uyuni.x86_64'
copy 'python-msgpack-python-0.4.6-8.5.7.uyuni.x86_64'
copy 'python-psutil-5.2.2-15.8.6.uyuni.x86_64'
copy 'python-pycrypto-2.6.1-10.8.15.uyuni.x86_64'
copy 'python-PyYAML-5.1.2-26.14.3.uyuni.x86_64'
copy 'python-pyzmq-14.0.0-9.5.6.uyuni.x86_64'
copy 'python-requests-2.11.1-6.30.22.uyuni.noarch'
copy 'python-simplejson-3.8.2-14.21.uyuni.x86_64'
copy 'python-singledispatch-3.4.0.3-1.1.uyuni.noarch'
copy 'python-tornado-4.2.1-17.5.7.uyuni.x86_64'
copy 'salt-3000-46.106.1.uyuni.x86_64'
copy 'salt-minion-3000-46.106.1.uyuni.x86_64'
copy 'python2-salt-3000-46.106.1.uyuni.x86_64'
copy 'libgio-2_0-0-2.48.2-12.15.1.x86_64'
copy 'libgthread-2_0-0-2.48.2-12.15.1.x86_64'
copy 'shared-mime-info-1.6-11.3.x86_64'
copy 'glib2-tools-2.48.2-12.15.1.x86_64'
copy 'libelf0-0.8.13-18.64.x86_64'
copy 'logrotate-3.11.0-2.14.1.x86_64'
copy 'cron-4.2-59.13.1.x86_64'
copy 'cronie-1.4.11-59.13.1.x86_64'
copy 'libzmq3-4.0.4-15.5.12.uyuni.x86_64'
copy 'gio-branding-SLE-12-14.5.noarch'
copy 'wallpaper-branding-SLE-12-13.3.1.noarch'
copy 'libyui-ncurses-pkg7-2.48.4.1-1.5.x86_64'
copy 'libyui-qt-pkg7-2.45.13.1-1.6.x86_64'
copy 'python-enum34-1.1.3-8.3.3.noarch'
copy 'python-idna-2.5-3.10.2.noarch'
copy 'python-ipaddress-1.0.18-3.13.1.noarch'
copy 'python-setuptools-40.6.2-4.12.23.noarch'
copy 'dmidecode-3.0-10.3.1.x86_64'
Directory walk started
Directory walk done - 83 packages
Temporary output repo path: /srv/www/htdocs/pub/repositories/sle/12/4/bootstrap.tmp/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished
```

Bootstrapping the client fails with the following errors in Tomcat log:

```
2020-11-12 09:05:36,750 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-2] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - States failed during bootstrap: file_|-/etc/salt/minion.d/susemanager.conf_|-/etc/salt/minion.d/susemanager.conf_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
file_|-/etc/salt/pki/minion/minion.pem_|-/etc/salt/pki/minion/minion.pem_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
file_|-/etc/salt/pki/minion/minion.pub_|-/etc/salt/pki/minion/minion.pub_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
service_|-salt-minion_|-salt-minion_|-running(retcode=2): One or more requisite failed: bootstrap./etc/salt/pki/minion/minion.pem, bootstrap.salt-minion-package, bootstrap./etc/salt/minion.d/susemanager.conf, bootstrap./etc/salt/pki/minion/minion.pub, bootstrap./etc/salt/minion_id
pkg_|-salt-minion-package_|-salt-minion_|-latest(retcode=2): No information found for 'salt-minion'.
file_|-/etc/salt/minion_id_|-/etc/salt/minion_id_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
2020-11-12 09:08:11,322 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-7] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - States failed during bootstrap: file_|-/etc/salt/minion.d/susemanager.conf_|-/etc/salt/minion.d/susemanager.conf_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
pkg_|-salt-minion-package_|-salt-minion_|-latest(retcode=2): No information found for 'salt-minion'.
file_|-/etc/salt/pki/minion/minion.pub_|-/etc/salt/pki/minion/minion.pub_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
file_|-/etc/salt/pki/minion/minion.pem_|-/etc/salt/pki/minion/minion.pem_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
file_|-/etc/salt/minion_id_|-/etc/salt/minion_id_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
service_|-salt-minion_|-salt-minion_|-running(retcode=2): One or more requisite failed: bootstrap.salt-minion-package, bootstrap./etc/salt/pki/minion/minion.pem, bootstrap./etc/salt/pki/minion/minion.pub, bootstrap./etc/salt/minion.d/susemanager.conf, bootstrap./etc/salt/minion_id
2020-11-12 09:12:31,336 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-2] ERROR com.suse.manager.webui.controllers.utils.AbstractMinionBootstrapper - States failed during bootstrap: file_|-/etc/salt/minion.d/susemanager.conf_|-/etc/salt/minion.d/susemanager.conf_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
service_|-salt-minion_|-salt-minion_|-running(retcode=2): One or more requisite failed: bootstrap./etc/salt/minion_id, bootstrap./etc/salt/pki/minion/minion.pem, bootstrap.salt-minion-package, bootstrap./etc/salt/pki/minion/minion.pub, bootstrap./etc/salt/minion.d/susemanager.conf
file_|-/etc/salt/minion_id_|-/etc/salt/minion_id_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
file_|-/etc/salt/pki/minion/minion.pub_|-/etc/salt/pki/minion/minion.pub_|-managed(retcode=2): One or more requisite failed: bootstrap.salt-minion-package
```

On the affect machine, I cannot see that the zypper bootstrap repository was added.

**Comment by mcalmer** [comment]
For better debugging what happen on the minion during the salt ssh session you can set the following on the **master**

/etc/salt/master.d/debug.conf
```
ssh_minion_opts:
   log_file: ../../../../../var/log/salt-ssh.log
   log_level: debug
```

restart salt-master/api and bootstrap again.
When it failed again check on the **minion** /var/log/salt-ssh.log

**Comment by stdevel** [comment]
Hi @mcalmer!
I will keep you updated with the information requested - I'm currently on vacation and have no access to the system.

Is this okay?

Best wishes!

**Comment by mcalmer** [comment]
@stdevel sure. I am out as well
Happy holidays

**Comment by Shirocco88** [comment]
@stdevel 
> I'm trying to bootstrap a SLES4SAP12 SP5 client with the UI
...
```
#  mgr-create-bootstrap-repo --with-custom-channels
...
Creating bootstrap repo for SLES4SAP-12-SP4-x86_64
```

Question: Is it correct, that you want to bootstrap a SLES4SAP-12 SP5 Client with SP4 salt-minion ?

**Comment by Shirocco88** [comment]
@stdevel : Uyuni-2021.05 was released;
is this issue fixed with uyuni-2021.05 ? can we close this issue ?

**Comment by stdevel** [comment]
Hi @Shirocco88 - sorry for the late answer. I don't have access to the systems to reproduce this issue anymore.
So we can close this issue - if I stumble upon this issue once again I would open another ticket.

Best wishes and thanks a lot for your efforts,
Christian.
